cfe_internal/core/watchdog/templates/watchdog.mustache

```diff
@@ -130,10 +130,27 @@ if [ "${PATHOLOGY_COUNT}" -gt "${PATHOLOGOY_THRESHOLD}" ]; then
     tar -c -f "${COLLECTION_DIR}/sys.workdir-before-remediation.tar.gz" -C "${CFENGINE_WORKDIR}" state outputs
     echo "- [[./sys.workdir-before-remediation.tar.gz][=sys.workdir-before-remediation.tar.gz=]] - CFEngine WORKDIR artifacts before remediation" >> ${COLLECTION_REPORT}
 
+    ps -efl | grep cf- > "${COLLECTION_DIR}/cf-procs.txt"
+    echo "- [[./cf-procs.txt][~ps -efl | grep cf-~]]" >> ${COLLECTION_REPORT}
+
+    while IFS= read -r proc
+    do
+        _PID=$(echo $proc | awk '{print $4}')
+        _PROC=$(echo $proc | awk '{print $15}')
+        _PROCFILE=$(echo "$_PROC" | sed 's./._.g')
+        _COREFILE=$(printf "%s_%s.core" "$_PID" "$_PROCFILE")
+        _COREPATH=$(printf "%s/%s" "${COLLECTION_DIR}" "$_COREFILE")
+        _DBX_ERR_LOG=$(printf "%s/%s.dbx.err" "${COLLECTION_DIR}" "$_COREFILE")
+
+        gencore "$_PID" "$_COREPATH"
+        echo "- [[./$_COREFILE][gencore $_PID $_COREPATH]]" >> ${COLLECTION_REPORT}
+
+        observation=$(echo "where" | dbx "$_PROC" "$_COREPATH" 2>"$_DBX_ERR_LOG")
+        echo "  - backtrace:" >> ${COLLECTION_REPORT}
+        echo "    #+begin_example\n${observation}\n#+end_example" >> ${COLLECTION_REPORT}
+        echo "  - dbx stderr: [[$_DBX_ERR_LOG]]" >> ${COLLECTION_REPORT}
+    done < "${COLLECTION_DIR}/cf-procs.txt"
```

Comment:
after we coredump and backtrace each PID we could kill it. @nickanderson sound good?

Reply:
you added this.